From 49f0271202d9da5f99049ef7befef499dda954e7 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Thu, 17 May 2018 23:25:13 +0800
Subject: [PATCH 3822/5242] MLK-18334: regulator: bd71837: fix reboot failure
 on i.mx845s evk

commit  6cb4edc77e6c3e04ca7799be5fdebe567ce11895 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix reboot failure on i.mx845s evk board. For BD71837, BUCKx_SEL and
BUCKx_EN of all bucks can't be set to 1 as explicity enabled or disabled,
otherwise, it may mess up with the default state machine if pmic reset
triggered. Remove enable/disable interfaces in driver to avoid touch such
bits.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/bd71837-regulator.c |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/regulator/bd71837-regulator.c b/drivers/regulator/bd71837-regulator.c
index 4b04fb1..a4cd420 100644
--- a/drivers/regulator/bd71837-regulator.c
+++ b/drivers/regulator/bd71837-regulator.c
@@ -101,8 +101,6 @@ static int bd71837_buck1234_set_ramp_delay(struct regulator_dev *rdev, int ramp_
 };
 
 static struct regulator_ops bd71837_buck_regulator_ops = {
-	.enable = regulator_enable_regmap,
-	.disable = regulator_disable_regmap,
 	.is_enabled = regulator_is_enabled_regmap,
 	.list_voltage = regulator_list_voltage_linear_range,
 	.set_voltage_sel = regulator_set_voltage_sel_regmap,
@@ -111,8 +109,6 @@ static int bd71837_buck1234_set_ramp_delay(struct regulator_dev *rdev, int ramp_
 };
 
 static struct regulator_ops bd71837_buck1234_regulator_ops = {
-	.enable = regulator_enable_regmap,
-	.disable = regulator_disable_regmap,
 	.is_enabled = regulator_is_enabled_regmap,
 	.list_voltage = regulator_list_voltage_linear_range,
 	.set_voltage_sel = regulator_set_voltage_sel_regmap,
-- 
1.7.9.5

